pkg/ddl/backfilling_scheduler.go

```diff
@@ -422,7 +422,15 @@ func (b *ingestBackfillScheduler) setupWorkers() error {
 	for _, e := range b.reorgInfo.elements {
 		indexIDs = append(indexIDs, e.ID)
 	}
-	engines, err := b.backendCtx.Register(indexIDs, job.TableName)
+	var uniques []bool
+	switch v := job.Args[0].(type) {
+	case *bool:
```

Comment:
`*bool` or `bool`?

Reply:
it's really `*bool` I don't know why 😂

Reply:
very hard to know the exact type of different DDL jobs, hope we can improve it in DDL refactor soon.

Reply:
`TestGlobalSortMultiSchemaChange` failed, that's why I have this question..

```
[2024/05/31 03:39:12.035 +00:00] [ERROR] [misc.go:114] ["panic in the recoverable goroutine"] [label=ddl] [funcInfo=onCreateIndex] [r="runtime error: index out of range [0] with length 0"] [stack="github.com/pingcap/tidb/pkg/util.Recover\n\tpkg/util/misc.go:118\nruntime.gopanic\n\tGOROOT/src/runtime/panic.go:914\nruntime.goPanicIndex\n\tGOROOT/src/runtime/panic.go:114\ngithub.com/pingcap/tidb/pkg/ddl/ingest.(*litBackendCtx).Register\n\tpkg/ddl/ingest/engine_mgr.go:80\ngithub.com/pingcap/tidb/pkg/ddl.(*ingestBackfillScheduler).setupWorkers\n\tpkg/ddl/backfilling_scheduler.go:433\ngithub.com/pingcap/tidb/pkg/ddl.(*ddlCtx).writePhysicalTableRecord\n\tpkg/ddl/backfilling.go:602\ngithub.com/pingcap/tidb/pkg/ddl.(*worker).addPhysicalTableIndex\n\tpkg/ddl/index.go:1950\ngithub.com/pingcap/tidb/pkg/ddl.(*worker).addTableIndex\n\tpkg/ddl/index.go:1997\ngithub.com/pingcap/tidb/pkg/ddl.runReorgJobAndHandleErr.func1\n\tpkg/ddl/index.go:1056\ngithub.com/pingcap/tidb/pkg/ddl.(*worker).runReorgJob.func1\n\tpkg/ddl/reorg.go:254"] [stack="github.com/pingcap/tidb/pkg/util.Recover\n\tpkg/util/misc.go:114\nruntime.gopanic\n\tGOROOT/src/runtime/panic.go:914\nruntime.goPanicIndex\n\tGOROOT/src/runtime/panic.go:114\ngithub.com/pingcap/tidb/pkg/ddl/ingest.(*litBackendCtx).Register\n\tpkg/ddl/ingest/engine_mgr.go:80\ngithub.com/pingcap/tidb/pkg/ddl.(*ingestBackfillScheduler).setupWorkers\n\tpkg/ddl/backfilling_scheduler.go:433\ngithub.com/pingcap/tidb/pkg/ddl.(*ddlCtx).writePhysicalTableRecord\n\tpkg/ddl/backfilling.go:602\ngithub.com/pingcap/tidb/pkg/ddl.(*worker).addPhysicalTableIndex\n\tpkg/ddl/index.go:1950\ngithub.com/pingcap/tidb/pkg/ddl.(*worker).addTableIndex\n\tpkg/ddl/index.go:1997\ngithub.com/pingcap/tidb/pkg/ddl.runReorgJobAndHandleErr.func1\n\tpkg/ddl/index.go:1056\ngithub.com/pingcap/tidb/pkg/ddl.(*worker).runReorgJob.func1\n\tpkg/ddl/reorg.go:254"]
```

Reply:
I'll add another case. `*bool` type is also chosen from a test case.

Reply:
https://github.com/pingcap/tidb/pull/53632/commits/f94911d1e05e33aeab1a5234ec1d34fd23f187d7